### ISSUE TITLE
back-transformation giving very small se for GPP

### ISSUE THREAD

**Issue by fawda123** [tbep-tech/wqtrends#2]
Estimates of seasonal averages for GPP give very small standard error estimates on the back-transformed results.  Works fine for chl:

```
library(wqtrends)
library(dplyr)
library(ggplot2)

tomod <- rawdat %>%
  filter(station %in% 32) %>%
  filter(param %in% 'chl')

gam2 <- anlz_gam(tomod, mod = 'gam2', trans = 'log10')
  
avgs <- anlz_avgseason(mods = list(gam2), doystr = 90, doyend = 180)

ggplot(avgs, aes(x = yr, y = predicted)) +
  geom_point() + 
  geom_errorbar(aes(ymin = predicted - se, ymax = predicted + se))
```
![image](https://user-images.githubusercontent.com/3086881/95657418-6ba82380-0ae2-11eb-9435-331b62e86038.png)

GPP gives very small standard error:

```
tomod <- rawdat %>%
  filter(station %in% 32) %>%
  filter(param %in% 'gpp')

gam2 <- anlz_gam(tomod, mod = 'gam2', trans = 'log10')
  
avgs <- anlz_avgseason(mods = list(gam2), doystr = 90, doyend = 180)

ggplot(avgs, aes(x = yr, y = predicted)) +
  geom_point() + 
  geom_errorbar(aes(ymin = predicted - se, ymax = predicted + se))
```
![image](https://user-images.githubusercontent.com/3086881/95657436-88dcf200-0ae2-11eb-8523-970bca545ae5.png)

GPP using identity transformation looks better. 
```
tomod <- rawdat %>%
  filter(station %in% 32) %>%
  filter(param %in% 'gpp')

gam2 <- anlz_gam(tomod, mod = 'gam2', trans = 'ident')
  
avgs <- anlz_avgseason(mods = list(gam2), doystr = 90, doyend = 180)

ggplot(avgs, aes(x = yr, y = predicted)) +
  geom_point() + 
  geom_errorbar(aes(ymin = predicted - se, ymax = predicted + se))
```
![image](https://user-images.githubusercontent.com/3086881/95657482-b033bf00-0ae2-11eb-869e-577c58ad3db7.png)

The small CI values occur for both BoxCox and log transformations of GPP.  It is likely an error with the back-transformation bias correction factor starting here: https://github.com/tbep-tech/wqtrends/blob/1b5f80924b6c094ae3049a836627b77ad5d584fc/R/anlz_avgseason.R#L99

**Comment by fawda123** [comment]
See email from PDV 11/5

**Comment by fawda123** [comment]
Fixed w/ commit ending 6486c20dcd4a1db6c30bc4f91839071b88a51f78

```
library(wqtrends)
library(dplyr)
library(ggplot2)

tomod <- rawdat %>%
  filter(station %in% 34) %>%
  filter(param %in% 'gpp')

mod <- anlz_gam(tomod, trans = 'log10')

avgs <- anlz_avgseason(mod, doystr = 90, doyend = 180)

ggplot(avgs, aes(x = yr, y = avg)) +
  geom_point() + 
  geom_errorbar(aes(ymin = lwr, ymax = upr))
```

![image](https://user-images.githubusercontent.com/3086881/99190877-32914d80-2737-11eb-87e0-5c136aafb875.png)

And identity...

```
mod <- anlz_gam(tomod, trans = 'identity')

avgs <- anlz_avgseason(mod, doystr = 90, doyend = 180)

ggplot(avgs, aes(x = yr, y = avg)) +
  geom_point() + 
  geom_errorbar(aes(ymin = lwr, ymax = upr))
```

![image](https://user-images.githubusercontent.com/3086881/99190918-6f5d4480-2737-11eb-8c87-59857f68aa99.png)

Fixes are here: 
https://github.com/tbep-tech/wqtrends/blob/d7d27c7639835785466933434d3562b73bf815b1/R/anlz_avgseason.R#L70
https://github.com/tbep-tech/wqtrends/blob/d7d27c7639835785466933434d3562b73bf815b1/R/anlz_perchg.R#L97
